contracts/zero-ex/contracts/src/features/MultiplexFeature.sol

```diff
@@ -0,0 +1,799 @@
+                    // msg.sender, so we must transfer the input token
+                    // to the FlashWallet here.
+                    _transferERC20Tokens(
+                        args.inputToken,
+                        msg.sender,
+                        flashWallet,
+                        outputTokenAmount
+                    );
+                    args.inputTokenAmount = 0;
+                } else {
+                    // Otherwise, either:
+                    // (1) args.taker == msg.sender, in which case
+                    //     `_transformERC20` will pull the input token
+                    //     into the FlashWallet, or
+                    // (2) args.inputToken == ETH_TOKEN_ADDRESS, in which
+                    //     case ETH is attached to the call and no token
+                    //     transfer occurs.
+                    args.inputTokenAmount = outputTokenAmount;
+                }
+                uint256 ethValue;
+                (args.transformations, ethValue) = abi.decode(
+                    wrappedCall.data,
+                    (ITransformERC20Feature.Transformation[], uint256)
+                );
+                // Do not spend more than the remaining ETH.
+                ethValue = LibSafeMathV06.min256(ethValue, remainingEth);
+                // Call `_transformERC20`.
+                outputTokenAmount = ITransformERC20Feature(address(this))
+                    ._transformERC20{value: ethValue}(args);
+                // Decrement the remaining ETH.
+                remainingEth -= ethValue;
+                // If the recipient was not `msg.sender`, it must be the target
+                // contract for the next hop.
+                nextTarget = args.taker == msg.sender ? address(0) : args.taker;
+            } else if (wrappedCall.selector == this._batchFill.selector) {
+                BatchFillData memory batchFillData;
+                batchFillData.inputToken = IERC20TokenV06(fillData.tokens[i]);
+                batchFillData.outputToken = IERC20TokenV06(fillData.tokens[i + 1]);
+                uint256 ethValue;
+                (batchFillData.calls, ethValue) = abi.decode(
+                    wrappedCall.data,
+                    (WrappedBatchCall[], uint256)
+                );
+                // Do not spend more than the remaining ETH.
+                ethValue = LibSafeMathV06.min256(ethValue, remainingEth);
+                if (IERC20TokenV06(fillData.tokens[i]).isTokenETH()) {
+                    batchFillData.sellAmount = LibSafeMathV06.min256(
+                        outputTokenAmount,
+                        ethValue
+                    );
+                } else {
+                    batchFillData.sellAmount = outputTokenAmount;
+                }
+                // Decrement the remaining ETH.
+                remainingEth -= ethValue;
+                uint256 leftoverEth;
+                (outputTokenAmount, leftoverEth) = _batchFill(batchFillData, ethValue);
```

Comment:
![13170221](https://user-images.githubusercontent.com/48803443/108753198-6b994500-7512-11eb-8169-26d7974e2915.jpg)
Surprised this even works.